src/Microsoft.DncEng.PatGeneratorTool/Program.cs

```diff
@@ -130,7 +132,16 @@ private static async Task<int> Go(List<AzureDevOpsPATScopes> scopes, string[] or
                     user = $"{user.Split('\\').Last()}@microsoft.com";
                 }
 
-                credentials = new VssAadCredential(user, password);
+                var app = PublicClientApplicationBuilder
+                    .Create(AzdoClientId)
+                    .WithAuthority("https://login.microsoftonline.com/microsoft.com/")
+                    .Build();
+
+                var passwordSecureStr = new SecureString();
+                Array.ForEach(password!.ToCharArray(), passwordSecureStr.AppendChar);
+
+                var authResult = await app.AcquireTokenByUsernamePassword(AzureDevOpsAuthScopes, user, passwordSecureStr).ExecuteAsync();
```

Comment:
I notice there is an "AquireTokenInteractive"? Does that actually work? Because if so, that's way better than requiring a usename/password on the command line, and we should support that.

Reply:
Ah, seems like it needs some sort of localhost thing listening to gather the garbage. That's not cool.

Reply:
Oh! No, it worked if I added `WithRedirectUri("http://localhost")` to the URI.

Reply:
This could actually really be useful for a lot of tools... being able to do interact login with .NET Core means we can write tools that muck around with Azure DevOps as the current user without having to make things .NET Framework apps.

Reply:
This thing can do interactive login for core already, see

https://github.com/dotnet/arcade-services/blob/38ce434f2d8df12fe0632211152b1247d0edaab3/src/Microsoft.DncEng.PatGeneratorTool/Program.cs#L212-L215

Reply:
That works?!?!? Where was this when I needed it. :-) Seems like we should use the same library for both things and update this one too then.

Reply:
I had no idea you could just pass a random Azure token into Azure DevOps as a basic cred... that makes... like negative sense. But it works, I guess?

Reply:
That's the approved form:

https://docs.microsoft.com/en-us/rest/api/azure/devops/?view=azure-devops-rest-7.1

```
public static async void GetProjects()
{
	try
	{
		var personalaccesstoken = "PAT_FROM_WEBSITE";

		using (HttpClient client = new HttpClient())
		{
			client.DefaultRequestHeaders.Accept.Add(
				new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));

			client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
				Convert.ToBase64String(
					System.Text.ASCIIEncoding.ASCII.GetBytes(
						string.Format("{0}:{1}", "", personalaccesstoken))));

			using (HttpResponseMessage response = await client.GetAsync(
						"https://dev.azure.com/{organization}/_apis/projects"))
			{
				response.EnsureSuccessStatusCode();
				string responseBody = await response.Content.ReadAsStringAsync();
				Console.WriteLine(responseBody);
			}
		}
	}
	catch (Exception ex)
	{
		Console.WriteLine(ex.ToString());
	}
}
```